cmd/kube-controller-manager/app/plugins.go

```diff
@@ -66,9 +70,36 @@ func ProbeRecyclableVolumePlugins(flags VolumeConfigFlags) []volume.VolumePlugin
 	}
 	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
 
+	allPlugins = append(allPlugins, aws_ebs.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, gce_pd.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, cinder.ProbeVolumePlugins()...)
+
 	return allPlugins
 }
 
+// NewVolumeProvisionersForCloud maps a cloud provider to a specific volume plugin.
+func NewVolumeProvisioners(plugins []volume.VolumePlugin, qosClasses []string) map[string]volume.ProvisionableVolumePlugin {
+	provisioners := map[string]volume.ProvisionableVolumePlugin{}
+	for _, qos := range qosClasses {
+		// the value is "key/value" and requires parsing after the first slash.
+		// values will be volume plugin names, many of which also contain slashes in the name.
+		firstSlash := strings.Index(qos, "/")
+		// 0 cannot be the first slash. there would be no tier name.
+		if firstSlash > 0 {
+			tier := qos[0:firstSlash]
+			provisioner := qos[(firstSlash + 1):] // needs +1 because we don't want the slash prefix
+			for _, plugin := range plugins {
+				if plugin.Name() == provisioner {
+					if provisonablePlugin, ok := plugin.(volume.ProvisionableVolumePlugin); ok {
+						provisioners[tier] = provisonablePlugin
+					}
+				}
```

Comment:
Agreed.  Failing silently means a cluster won't behave as expected.  Better to fail loudly and remove ambiguity.